src/Akeneo/Pim/Structure/Bundle/Query/PublicApi/Family/Sql/SqlGetRequiredAttributesMasks.php

```diff
@@ -17,98 +16,40 @@
-                        FROM pim_catalog_channel channel
-                        JOIN pim_catalog_channel_currency pccc ON channel.id = pccc.channel_id
-                        JOIN pim_catalog_currency currency ON pccc.currency_id = currency.id
-                        WHERE channel.code  = channel_code
-                        GROUP BY channel.id
-                    )
-                ),
-                attribute.code
-            ),
-            '-',
-            IF(attribute.is_scopable, channel_locale.channel_code, '<all_channels>'),
-            '-',
-            IF(attribute.is_localizable, channel_locale.locale_code, '<all_locales>')
-        )
-    ) AS mask
-FROM pim_catalog_family family
-JOIN pim_catalog_attribute_requirement pcar ON family.id = pcar.family_id
-JOIN (
-    SELECT
-        channel.id AS channel_id,
-        channel.code AS channel_code,
-        locale.id AS locale_id,
-        locale.code AS locale_code
-    FROM pim_catalog_channel channel
-    JOIN pim_catalog_channel_locale pccl ON channel.id = pccl.channel_id
-    JOIN pim_catalog_locale locale ON pccl.locale_id = locale.id
-) AS channel_locale ON channel_locale.channel_id = pcar.channel_id
-JOIN pim_catalog_attribute attribute ON pcar.attribute_id = attribute.id
-LEFT JOIN pim_catalog_attribute_locale pcal ON attribute.id = pcal.attribute_id
-WHERE
-    pcar.required is true
-    AND (pcal.locale_id IS NULL OR pcal.locale_id = channel_locale.locale_id)
-    AND family.code IN (:familyCodes)
-GROUP BY family.code, channel_code, locale_code
-SQL;
-
-        $rows = $this->connection->executeQuery(
-            $sql,
-            ['familyCodes' => $familyCodes],
-            ['familyCodes' => Connection::PARAM_STR_ARRAY]
-        )->fetchAllAssociative();
-
-        $families = array_map(function (array $row): string {
-            return $row['family_code'];
-        }, $rows);
+        $result = [];
+        $foundFamilyCodes = [];
+        foreach ($this->getAttributeMasksPerAttributeTypes as $getAttributeMasksPerAttributeType) {
+            $masksPerAttributeTypeCollection = $getAttributeMasksPerAttributeType->fromFamilyCodes($familyCodes);
+            foreach ($masksPerAttributeTypeCollection as $familyCode => $requiredAttributeMask) {
+                array_push($foundFamilyCodes, $familyCode);
+                if (!isset($result[$familyCode])) {
+                    $result[$familyCode] = $requiredAttributeMask;
+                } else {
+                    $formerMasks = $result[$familyCode]->masks();
+                    $masksForThisAttributeType = $requiredAttributeMask->masks();
+                    $result[$familyCode] = new RequiredAttributesMask($familyCode, array_merge($formerMasks, $masksForThisAttributeType));
+                }
+            }
+        }
 
-        $nonExistingFamilyCodes = array_diff($familyCodes, $families);
+        $nonExistingFamilyCodes = array_diff($familyCodes, $foundFamilyCodes);
```

Comment:
```suggestion
        $nonExistingFamilyCodes = \array_diff($familyCodes, \array_keys($result));
```

And $foundFamilyCodes can be removed no?